### ISSUE TITLE
traitlets v4.2.0 issues warnings upon startup

### ISSUE THREAD

**Issue by brbsix** [ipython/ipython#9324]
I receive the following messages upon startup of IPython 4.1.2 in Python 3.4.3.

```
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(UserMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(AutoMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(BasicMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(CodeMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(ConfigMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(DeprecatedMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(DisplayMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(ExecutionMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(ExtensionMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(HistoryMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(LoggingMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(NamespaceMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(OSMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(PylabMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(ScriptMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(TerminalMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)
Python 3.4.3 (default, Mar 26 2015, 22:03:40) 
Type "copyright", "credits" or "license" for more information.

IPython 4.1.2 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
/home/user/.local/lib/python3.4/site-packages/traitlets/config/configurable.py:74: RuntimeWarning: Passing unrecoginized arguments to super(StoreMagics).__init__(shell=<IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x7f852c2c83c8>).
object.__init__() takes no parameters
This error will be raised in a future release of traitlets.
  super(Configurable, self).__init__(**kwargs)

```

I noticed this behavior after updating traitlets to version 4.2.0.

It seems to be the result of https://github.com/ipython/traitlets/commit/86c6efa60b79d49e2a65fcee2678e99b60741d02. There's an entry in the [changelog](https://github.com/ipython/traitlets/blob/master/docs/source/changelog.rst) about it as well:

> `Unused arguments are passed to super() in HasTraits.__init__, improving support for multiple inheritance.`

**Comment by jason-womack** [comment]
I am seeing this too. 
IPython 4.1.2
Python 3.5.1

**Comment by minrk** [comment]
These have been fixed by #9238

**Comment by minrk** [comment]
I'll demote this warning to DeprecationWarning so you won't see them by default, and release 4.2.1 shortly.

**Comment by minrk** [comment]
Just released traitlets-4.2.1, which should fix this.
